src/index.js

```diff
@@ -1,190 +1,56 @@
-        await new Promise(resolve => {
-            this.room.on('subscribed', () => resolve());
-        });
-        // If customer, also wait for at least one peer to join the room (TODO: review this)
-        if (this.type === constant.types.customer) {
-            await new Promise(resolve => {
-                this.room.on('peer joined', () => resolve());
-            });
-        }
-        return true;
+        return await this.instance.start();
     }
 
     async stop() {
-        await this.room.leave();
-        if (this.db) await this.db.close();
-        await this.ipfs.stop();
-        this.db = undefined;
-        this.room = undefined;
-        this.orbitDb = undefined;
-        this.ipfs = undefined;
+        return await this.instance.stop();
     }
 
     async pin(multihash){
-        // broadcast start of pin process
-        await this.room.broadcast(this.utils.encodeMessage({ type: constant.eventTypes.pinning, multihash }));
-        if (this.ipfs.pin) {
-            await this.ipfs.pin.add(multihash, { recursive: true });
-        } else {
-            // TODO: Bad!!! fix!!!!
-            console.warn('This node is running an IPFS client that does not implement pinning. Falling back to just retrieving the data non recursively');
-            await this.ipfs.object.get(multihash);
-        }
-        // broadcast successful pin
-        await this.room.broadcast(this.utils.encodeMessage({ type: constant.eventTypes.pinned, multihash }));
-    }
-
-    getOrbitDBAddress(){
-        if (this.type === constant.types.customer) {
-            return this.db.address.toString();
-        } else {
-            throw new Error('Not a customer');
-        }
+        return await this.instance.pin(multihash);
     }
 
     async storeReviewRecord(reviewRecord){
-        // TODO: check format, check is customer
-        // write thing to ipfs
-        const dagNode = await this.ipfs.object.put(reviewRecord);
-        const multihash = this.utils.multihashToString(dagNode.multihash);
-        // Broadcast request for pin, then wait for response
-        await new Promise(fullfill => {
-            this.events.once(constant.eventTypes.pinned + '_' + multihash, () => fullfill());
-            this.room.broadcast(this.utils.encodeMessage({ type: constant.eventTypes.wroteReviewRecord, multihash }));
-        });
-        return multihash;
+        return await this.instance.storeReviewRecord(reviewRecord);
     }
 
     async exportData() {
-        const exported = {};
-        if (this.type === constant.types.customer) {
-            exported.customerDbKeys = {
-                pub: await this.db.keystore.exportPublicKey(),
-                priv: await this.db.keystore.exportPrivateKey()
-            };
-        }
-        return exported;
+        return await this.instance.exportData();
     }
 
-    async importData() {
-        throw new Error('not implemented');
+    async importData(exportedData) {
```

Comment:
what data do exportData and importData do? Maybe we can change these function names to more meaningful or descriptive names?

Reply:
Yes that's a good idea, but first I need to figure out a good format for exporting what we need.

Both OrbitDB and IPFS have almost no support for exporting and importing keys so this could require some hacks, although I'm confident it's possible